src/DotNetWorker/Hosting/GrpcServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Channels;
+using Grpc.Core;
+using Grpc.Net.Client;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Diagnostics;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using static Microsoft.Azure.Functions.Worker.Grpc.Messages.FunctionRpc;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    internal static class GrpcServiceCollectionExtensions
+    {
+        internal static IServiceCollection RegisterOutputChannel(this IServiceCollection services)
+        {
+            return services.AddSingleton<GrpcHostChannel>(s =>
+            {
+                UnboundedChannelOptions outputOptions = new UnboundedChannelOptions
+                {
+                    SingleWriter = false,
+                    SingleReader = true,
+                    AllowSynchronousContinuations = true
+                };
+
+                return new GrpcHostChannel(System.Threading.Channels.Channel.CreateUnbounded<StreamingMessage>(outputOptions));
+            });
+        }
+
+        public static IServiceCollection AddGrpc(this IServiceCollection services)
```

Comment:
The type is internal as additional work is required to expose this. Just flagging as we want this to be an option.